src/umpire/strategy/MixedPool.hpp

```diff
@@ -0,0 +1,80 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_MixedPool_HPP
+#define UMPIRE_MixedPool_HPP
+
+#include <memory>
+#include <array>
+#include <map>
+
+#include "umpire/strategy/AllocationStrategy.hpp"
+
+#include "umpire/strategy/DynamicPool.hpp"
+#include "umpire/strategy/FixedPool.hpp"
+
+#include "umpire/Allocator.hpp"
+
+namespace umpire {
+namespace strategy {
+
+/**
+ * \brief A faster pool that pulls from a series of pools
+ *
+ * Pool implementation using a series of FixedPools for small sizes,
+ * and a DynamicPool for sizes larger than (1 << LastFixed) bytes.
+ */
+template<int FirstFixed = 8, int Increment = 1, int LastFixed = 22>
```

Comment:
@jdahm, I would like to see a recipe for usage of the MixedPoolImpl that uses both the default values for FirstFixed, Increment, and LastFixed and values set by the caller.

Having a comment in the recipe above stating what setting those values will do will be helpful and will illustrate the flexibility of this pool.

Reply:
Even when we change to non-templated fixed pools we'll probably stick to powers of 2 for the sizes. This class won't be templated then, but will probably have a constructor that takes a `vector<int>` of those powers. I'll write one for this version, and we can change it later.